core-api/src/main/java/com/optimizely/ab/event/NoopEventHandler.java

```diff
@@ -29,7 +29,8 @@
     private static final Logger logger = LoggerFactory.getLogger(NoopEventHandler.class);
 
     @Override
-    public void dispatchEvent(String url, Map<String, String> params) {
-        logger.debug("Called dispatchEvent with URL: {} and params: {}", url, params);
+    public void dispatchEvent(LogEvent logEvent) {
```

Comment:
May be not change this signature and have `dispatchEvent` continue having the same signature, with the consumer providing this information. `dispatchEvent` should be as naive as possible with minimum overhead for anyone trying to define it. In this case, they would need to know the methods provided by `LogEvent` to be able to create `dispatchEvent`